### PR TITLE
GEODE-4248: Converts to/fromData params to ref from shared_ptr.

### PR DESCRIPTION
**Pull request by pivotal-jbarrett**

### CODE REVIEW

File: docs/geode-native-docs/cpp-caching-api/pdxserializable-interface.html.md.erb [hunk 1]
```diff
@@ -87,15 +87,15 @@ PdxObject::~PdxObject() {
 }
```
Comment (mcmellawatt) [inline docs/geode-native-docs/cpp-caching-api/pdxserializable-interface.html.md.erb]: Documentation file change not intentional

File: docs/geode-native-docs/cpp-caching-api/using-enum-type-with-pdx.html.md.erb [hunk 2]
```diff
@@ -62,13 +62,13 @@ enum enumQuerytest { id1, id2, id3 };
     PdxEnumTestClass() { }
```
Comment (mcmellawatt) [inline docs/geode-native-docs/cpp-caching-api/using-enum-type-with-pdx.html.md.erb]: Documentation file change not intentional

File: cppcache/include/geode/PdxWriter.hpp [hunk 3]
```diff
@@ -162,8 +158,8 @@ class CPPCACHE_EXPORT PdxWriter {
    * @throws IllegalStateException if the named field has already been written
    * or fieldName is nullptr or empty
    */
-  virtual std::shared_ptr<PdxWriter> writeDouble(const std::string& fieldName,
-                                                 double value) = 0;
+  virtual PdxWriter& writeDouble(const std::string& LfieldName,
```
[the remark this answers is not in the document]
Comment (pivotal-jbarrett) [reply cppcache/include/geode/PdxWriter.hpp]: fat finger!

File: cppcache/src/PdxHelper.cpp [hunk 4]
```diff
@@ -122,29 +121,28 @@ void PdxHelper::serializePdx(
...
-      auto mergedPdxType = pdxTypeRegistry->getPdxType(pd->getMergedTypeId());
-      prw = std::make_shared<PdxRemoteWriter>(output, mergedPdxType, pd,
-                                              pdxTypeRegistry);
-    } else {
-      prw =
-          std::make_shared<PdxRemoteWriter>(output, className, pdxTypeRegistry);
-    }
-    pdxObject->toData(std::dynamic_pointer_cast<PdxWriter>(prw));
-    prw->endObjectWriting();
+    auto createPdxRemoteWriter = [&]() -> PdxRemoteWriter {
```
[the remark this answers is not in the document]
Comment (pivotal-jbarrett) [reply cppcache/src/PdxHelper.cpp]: The type lacks a copy or move assignment function so no you can't declare the value and assign it later. The simple workaround is to use a function to construct and r-value. The easiest function is a lambda.

File: cppcache/src/PdxInstanceImpl.cpp [hunk 5]
```diff
@@ -1432,20 +1432,20 @@ PdxFieldTypes::PdxFieldType PdxInstanceImpl::getFieldType(
   return static_cast<PdxFieldTypes::PdxFieldType>(pft->getTypeId());
 }
 
-void PdxInstanceImpl::writeUnmodifieldField(
-    DataInput& dataInput, int startPos, int endPos,
-    std::shared_ptr<PdxLocalWriter> localWriter) {
+void PdxInstanceImpl::writeUnmodifieldField(DataInput& dataInput, int startPos,
+                                            int endPos,
```
[the remark this answers is not in the document]
Comment (pivotal-jbarrett) [reply cppcache/src/PdxInstanceImpl.cpp]: Yes, expected.

**Pull request merged**